docs/adr/006_cups.md

```diff
@@ -0,0 +1,285 @@
+
+The aim of this document section is to take and record a decision for:
+
+- Which authentication mode needs to be used for connecting to CUPS endpoint
+- Which authentication mode needs to be used for connecting to LNS endpoint
+- Where are the certificates stored and how to retrieve those?
+
+### Decision(s)
+
+Out of the four authentication modes we should aim for "TLS Server and Client Authentication" for both CUPS and LNS endpoints.
+
+Driving this decision is the willing of increasing the overall security of the system.
+
+In order to keep the starter kit simple, the same set of server and client credentials are to be used for both CUPS and LNS protocol endpoints.
+
+### Authentication to CUPS endpoint
+
+The CUPS endpoint should support mutual TLS authentication between client (basic station) and server (network server IoT Edge module).
+
+For increased security the server should "require" client authentication and verify the provided credentials.
+
+Therefore, in order for Basic Station to connect to the CUPS HTTPS endpoint the following files should be provided to the station:
+
+- cups.uri
+- cups.crt (including the certificate in PEM format)
+- cups.key (including the EC Private Key for the child certificate)
+- cups.trust (including the chain of trust certificate for the server-side certificate)
+
+You may find an example on how to generate all the needed certificates in [this script](https://github.com/lorabasics/basicstation/blob/master/examples/cups/prep.sh) from the Basic Station official GitHub repository.
+
+### Authentication to LNS endpoint
+
+As for CUPS, the LNS endpoint should support mutual TLS authentication between client (basic station) and server (network server IoT Edge module).
+
+For increased security the server should "require" client authentication and verify the provided credentials.
+
+Therefore, in order for Basic Station to connect to the LNS WSS endpoint the following files should be provided to the station:
+
+- tc.uri (i.e.: "wss://LoRaWanNetworkSrvModule:5001")
+- tc.crt (including the certificate in PEM format)
+- tc.key (including the EC Private Key for the child certificate)
+- tc.trust (including the chain of trust certificate for the server-side certificate)
+
+Same certificate files as above are expected to be used for the Starter Kit, even though you might want to consider increasing security of the solution by generating different certificates for this endpoint.
+
+There is no need to copy/paste manually these files in the Basic Station; the CUPS Protocol Implementation later described in this document will retrieve the needed credentials from a centralized repository and update the Basic Station configuration when it starts. This option is also providing an option to rotate certificates when these are near the expiration.
```

Comment:
How do we plan to invalidate the old credentials after rotation?

Reply:
I have just added some details of changes needed in "LoRa Device Provisioning CLI".

The idea, to be documented is the following:

- IT employee creates new client certificates for a concentrator
- IT employee uploads new certificates to concentrator
- IT employee issues an "update" command with the Device Provisioning CLI
- After verifying that new credentials were downloaded (automatically after 1day or by unplugging-replugging device for forcing CUPS), a "revoke" command needs to be issued with the Device Provisioning CLI to revoke old client certificate thumbprint

Anything that you would add here? @bastbu @Mandur

Reply:
I am not sure I follow the revoking part. As far as I understood as soon as we issue the update command the thumbprint in the desired properties would be updated to the new value?

Reply:
Also when we use the provisioning CLI to upload new certs I assumed we would be overwritting the old ones in the azure storage account, correct?

Reply:
Ok I understood after the discussion, the thumprints are in list to allow updates